feat: add support for async tokenSource and an optional callback for secretPublicKey

sakiv opened the pull request:
…etOrPublicKey.

<!-- Please only open PRs for issues in place. If there is no issue, please open the issue first. Then create this PR and link it to the issue by adding the issue number in the line below. -->

Closes #461

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

<!-- If you wrote meaningful commit messages, this is it. If you don't feel confident that your commit messages fully explain your changes and your reasoning behind it, then you can add more information to this PR. Or, even better, update your commit messages ;) -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dbartholomae/middy-middleware-jwt-auth?pullRequest=460) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/dbartholomae/middy-middleware-jwt-auth?pullRequest=460) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dbartholomae/middy-middleware-jwt-auth?pullRequest=460) it.</sub>

sakiv commented:
Hi @dbartholomae, any success on CLA issue?

sakiv commented:
Hi @dbartholomae, thanks for taking care of it. I have added sign-offs on my commits.

sakiv commented:
Hi @dbartholomae, I have fixed the code coverage issue. 

There was an unusual issue with Istanbul to report issue with async function. This seems to be a known issue. However to make sure we have code coverage, I have refactored the code. Please review the changes.
